subprojects/logging/src/integTest/groovy/org/gradle/DeprecationHandlingIntegrationTest.groovy

```diff
@@ -45,9 +80,9 @@ class DeprecatedTask extends DefaultTask {
         run()
 
         then:
-        output.contains("Build file '$buildFile': line 3")
+        output.contains("Build file '$buildFile': line 10")
```

Comment:
This is only happening because of the way the integration test is structured.

The message is triggered inside of a `.gradle` file which would not be the case in a real life situation since the code calling the `nagUserOfDiscontinuedMethod` would be in the Gradle codebase instead.

If `someFeature()` and `otherFeature()` were defined in either a Java or Groovy file then the result would be what you expected. As it is, `LoggingDeprecatedFeatureHandler.isGradleScriptElement` is returning true for line 10, so line 3, one stack element later, is not printed.

You can see that everything is working correctly in the new integration tests using the realistic build file you provided.